src/Features/CSharp/Portable/ConvertForEachToFor/CSharpConvertForEachToForCodeRefactoringProvider.cs

```diff
@@ -87,10 +88,11 @@ protected override (SyntaxNode start, SyntaxNode end) GetForEachBody(ForEachStat
             var collectionVariable = GetCollectionVariableName(
                 model, generator, foreachInfo, foreachCollectionExpression, cancellationToken);
 
-            var collectionStatementType = foreachInfo.RequireExplicitCastInterface
-                ? generator.GetTypeExpression(foreachInfo.Options, foreachInfo.ExplicitCastInterface) :
-                  generator.GetTypeExpression(foreachInfo.Options,
-                     model.GetTypeInfo(foreachCollectionExpression).Type ?? model.Compilation.GetSpecialType(SpecialType.System_Object));
+            var typeSymbol = foreachInfo.RequireExplicitCastInterface
+                ? foreachInfo.ExplicitCastInterface
+                : model.GetTypeInfo(foreachCollectionExpression).Type ?? model.Compilation.GetSpecialType(SpecialType.System_Object);
+
+            var collectionStatementType = typeSymbol.GenerateTypeSyntax();
```

Comment:
these are the places where i updated our features to now just use 'GenerateTypeSyntax()' confident that the simplifier would now use 'var' if htat's what the user wanted.